File: beads/beads.py
"""
ID: brkwok1
LANG: PYTHON3
TASK: beads
"""
import sys
sys.stderr.write('message')

with open('beads.in', 'r') as fin:
    N = int(fin.readline().strip())
    beads = fin.readline().strip()

    circle = beads + beads
    m = 0
    l = 0
    res = 0

    for i in range(1, len(circle)):
        l = i - 1
        r = i
        curr_left = circle[i - 1]
        curr_right = circle[i]

        while (r - l + 1) < N and l - 1 >= 0 and circle[l] == "w":
            l -= 1

        curr_left = circle[l]

        while (r - l + 1) < N and l - 1 >= 0 and (circle[l - 1] == "w" or circle[l - 1] == curr_left):
            l -= 1

        while (r - l + 1) < N and r + 1 < len(circle) and circle[r] == "w":
            r += 1

        curr_right = circle[r]

        while (r - l + 1) < N and r + 1 < len(circle) and (circle[r + 1] == "w" or circle[r + 1] == curr_right):
            r += 1

        res = max(res, r - l + 1)

    with open('beads.out', 'w') as fout:
        fout.write(str(res) + "\n")

File: beads/beads.cpp
/*
ID: brkwok1
LANG: C++11
TASK: beads
*/
#include <string>
#include <iostream>
#include <fstream>
#include <algorithm>
#include <set>
#include <vector>
#include <cassert>

using namespace std;

int main()
{
  ifstream fin("beads.in");
  ofstream fout("beads.out");
  int N, max = 0, streak, state, i, j;
  string s;
  char c;

  fin >> N >> s;
  s = s + s;

  for (i = 0; i < N; i++)
  {
    c = (char)s[i];

    if (c == 'w')
    {
      state = 0;
    }
    else
    {
      state = 1;
    }

    j = i;
    streak = 0;
    while (state <= 2)
    {
      while (j < N + i && (s[j] == c || s[j] == 'w'))
      {
        j++;
        streak++;
      }
      state++;
      c = s[j];
    }
    if (streak > max)
    {
      max = streak;
    }
  }

  fout << max << endl;

  return 0;
}